app/assets/javascripts/controllers/middleware_datasource/middleware_datasource_controller.js

```diff
@@ -119,15 +148,31 @@ function MwAddDatasourceCtrl($scope, $rootScope, miqService, mwAddDatasourceServ
   };
 
   vm.addDatasourceStep2Next = function() {
-    var useExistingDriver = vm.step2DsModel.selectedJdbcDriver !== '';
+    var selectedJdbcDriver = vm.step2DsModel.selectedJdbcDriver;
+    var useExistingDriver = selectedJdbcDriver !== '';
+    var dsSelection;
     vm.dsModel.step = 'STEP3';
+
     if (useExistingDriver) {
-      vm.step3DsModel.connectionUrl = mwAddDatasourceService.determineConnectionUrlFromExisting(vm.step2DsModel.selectedJdbcDriver);
+      dsSelection = mwAddDatasourceService.findDsSelectionFromDriver(selectedJdbcDriver);
+      vm.step3DsModel.connectionUrl = mwAddDatasourceService.determineConnectionUrlFromExisting(selectedJdbcDriver);
     } else {
-      vm.step3DsModel.connectionUrl = mwAddDatasourceService.determineConnectionUrl(vm.chooseDsModel.selectedDatasource);
+      dsSelection = vm.chooseDsModel.selectedDatasource;
+      vm.step3DsModel.connectionUrl = mwAddDatasourceService.determineConnectionUrl(dsSelection);
+    }
+    if(dsSelection.hasOwnProperty('properties')){
+      var i = 0;
+      for(var name in dsSelection.properties){
+        vm.step3DsModel.dsProps.push({id: i, name: name, value: dsSelection.properties[name]});
```

Comment:
Fixed. Not sure why I even used `for` normally I would use an iterator like `_each()`. Anyway, not going to change now that for diff comments.